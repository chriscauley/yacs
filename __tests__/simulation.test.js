import Simulation from '../client/simulation'

const DEFAULTS = {
  people: 200,
  infected: 1,
  lethality: 0.1,
  duration: 100,
  size: 500,
  radius: 5,
  dt: 0.1,
  seed: 12345,
}

test("Snapshot simulation", () => {
  // this is mostly to ensure that the seeded randomness is consistent
  const simulation = new Simulation({seed: 12345, people: 2})
  expect(simulation.pieces).toMatchSnapshot()
  simulation.step()
  simulation.step()
  expect(simulation.pieces).toMatchSnapshot()
})

test("Step speed test", () => {
  const simulation = new Simulation(DEFAULTS)
  let i = 1000
  while (i--) {
    simulation.step()
  }
})

test("Collide speed test", () => {
  const simulation = new Simulation(DEFAULTS)
  const p0 = simulation.pieces[0]
  const p1 = simulation.pieces[1]
  let i = 1000
  while (i--) {
    simulation.collide(p0, p1, 0)
  }
})