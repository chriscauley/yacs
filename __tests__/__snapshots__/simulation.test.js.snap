// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot simulation 1`] = `
Array [
  Object {
    "dx": 0.5035940373614989,
    "dy": -0.8639404178147618,
    "id": 1,
    "status": 0,
    "type": "person",
    "x": 473.85124883041834,
    "y": 17.93929726624796,
  },
  Object {
    "dx": 0.9973697847098382,
    "dy": 0.07248111856098155,
    "id": 2,
    "infected_until": 585.8699470626841,
    "status": 1,
    "type": "person",
    "x": 25.577609916755566,
    "y": 382.8935838610805,
  },
]
`;

exports[`Snapshot simulation 2`] = `
Array [
  Object {
    "dx": 0.5035940373614989,
    "dy": -0.8639404178147618,
    "id": 1,
    "status": 0,
    "type": "person",
    "x": 474.35484286777984,
    "y": 17.0753568484332,
  },
  Object {
    "dx": 0.9973697847098382,
    "dy": 0.07248111856098155,
    "id": 2,
    "infected_until": 585.8699470626841,
    "status": 1,
    "type": "person",
    "x": 26.574979701465406,
    "y": 382.9660649796415,
  },
]
`;
